homeassistant/helpers/template.py

```diff
@@ -916,6 +916,29 @@ def device_entities(hass: HomeAssistant, _device_id: str) -> Iterable[str]:
     return [entry.entity_id for entry in entries]
 
 
+def integration_entities(hass: HomeAssistant, entry_name: str) -> Iterable[str]:
+    """
+    Get entity ids for entities tied to an integration/domain.
+
+    Provide entry_name as domain to get all entity id's for a integration/domain
+    or provide a config entry title for filtering between instances of the same integration.
+    """
+    conf_entry = next(
+        (
+            entry.entry_id
+            for entry in hass.config_entries.async_entries()
+            if entry_name == entry.title
+        ),
+        None,
+    )
+    return [
+        entity_id
+        for entity_id, info in hass.data.get("entity_info", {}).items()
+        if (conf_entry is not None and info.get("config_entry") == conf_entry)
```

Comment:
Doing this all in in 1 list comprehension is a bit complicated, let's break it up.

Actually, can we just drop config entries and just do domain to start ?

Reply:
feedback I got from users is that only domain is not enough to grab specific entities e.g. for a specific Hue bridge or something similar.

Reply:
ok

Then still break up the list comprehension. Detect before you do it if it's a config entry check or domain check.

Reply:
OK, made some adjustments and splitted it up